src/providers/FundingEntityProvider.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { APP_PATHS } from 'src/constants';
+import FundingEntityService from 'src/services/FundingEntityService';
+import strings from 'src/strings';
+import { FundingEntity } from 'src/types/FundingEntity';
+import useEnvironment from 'src/utils/useEnvironment';
+
+import { ProvidedFundingEntityData } from './DataTypes';
+import { FundingEntityContext } from './contexts';
+import { useUser } from './hooks';
+
+export type FundingEntityProviderProps = {
+  children?: React.ReactNode;
+};
+
+enum APIRequestStatus {
+  'AWAITING',
+  'FAILED',
+  'FAILED_NO_AUTH',
+  'SUCCEEDED',
+}
+export default function FundingEntityProvider({ children }: FundingEntityProviderProps) {
+  const { user, bootstrapped: userBootstrapped } = useUser();
+  const [fundingEntity, setFundingEntity] = useState<FundingEntity | undefined | null>();
+  const [entityAPIRequestStatus, setEntityAPIRequestStatus] = useState<APIRequestStatus>(APIRequestStatus.AWAITING);
+  const navigate = useNavigate();
+  const { isDev, isStaging } = useEnvironment();
+
+  const populateFundingEntity = useCallback(async (userId: number) => {
+    const response = await FundingEntityService.getUserFundingEntity(userId);
```

Comment:
Could we do this using redux? I know we have some places where we still call services directly, but we are trying to use redux in new added code

Reply:
Ya I can update it to use redux. My redux skills are definitely weaker so I didn't think of that and was modeling all of this after Organization and User Providers which didn't seem to use it, but they're probably older code

Reply:
Yes, probably those were written before moving to redux

Reply:
Updated to redux!